### DISCUSSION TITLE
Image 2 Image is a pretty good partial upscaler

### DISCUSSION THREAD

**Discussion by seijihariki** [zero01101/openOutpaint#84] [Show and tell]
I have been testing the image 2 image tool on blurry scaled images, and it seems to work pretty well as an upscaler for parts of an image. Obviously, it also changes the image a bit, but that can be guided using the denoising strength and prompting.

Took around half an hour to do this one, but I was also experimenting some things ('sharp' in the prompt seems to help a lot), and was distracted by another thing that I was doing. I have done some 2048x2048 better than this one in less than 10 minutes.

Original (generated 512x512 and scaled to 2048x2048):

![2022-12-10 14 10 26 openOutpaint image](https://user-images.githubusercontent.com/7552751/206859599-8b10cbc3-5e83-4424-b1b1-3c4c2590630b.png)

Half-upscale (using cursors smaller than 2048x2048 - 1024x1024 and 1280x1280):

![2022-12-10 14 10 20 openOutpaint image](https://user-images.githubusercontent.com/7552751/206859659-e64f01d9-b21e-4d68-a8ee-2edf102335b5.png)

Final Upscale (using cursor of 512x512 or smaller):

![2022-12-10 14 10 09 openOutpaint image](https://user-images.githubusercontent.com/7552751/206859680-e9bce405-ace8-47c4-9d57-d856f1835d31.png)

A timelapse of the thing:
https://drive.google.com/file/d/1awkPnV6a-ziMBVlsYqCIORIYzbCTieOZ/view?usp=share_link

Any suggestions on how to improve results/more effective ways to do this are welcome!

**Comment by zero01101** [comment]
like i've definitely used img2img to result in """""[upscaling](https://twitter.com/djabacus/status/1573468208689971211)""""" but it was clearly 100% driven in tiny areas with much prompt assistance and took _hours_ lol


_oh my god_ you've got intermediary noise images displaying evolution in real time on some experimental branch and you just HAD to sneakily drop a quick preview didn't you 👀👀👀👀

[edit]

🤦‍♂️lmao welp TIL that "just works" if you enable the option in A1111 webUI which i'd left disabled basically forever but still, it seemed like a magic feature you'd just implement sneakily :D
